python/lsst/dax/apdb/sql/config.py

```diff
@@ -0,0 +1,154 @@
+# This file is part of dax_apdb.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["ApdbSqlConfig", "ApdbSqlConnectionConfig", "ApdbSqlPixelizationConfig"]
+
+from typing import Any, ClassVar
+
+from pydantic import BaseModel, Field, field_validator
+
+from ..config import ApdbConfig
+
+
+class ApdbSqlConnectionConfig(BaseModel):
+    """Configuration of connection parameters."""
+
+    isolation_level: str | None = Field(
+        default=None,
+        description=(
+            "Transaction isolation level, if unset then backend-default value "
+            "is used, except for SQLite backend where we use READ_UNCOMMITTED. "
+            "Some backends may not support every allowed value."
+        ),
+    )
+
+    connection_pool: bool = Field(
+        default=True,
+        description=(
+            "If False then disable SQLAlchemy connection pool. " "Do not use connection pool when forking."
+        ),
+    )
+
+    connection_timeout: float | None = Field(
+        default=None,
+        description=(
+            "Maximum time to wait time for database lock to be released before "
+            "exiting. Defaults to sqlalchemy defaults if not set."
+        ),
+    )
+
+    extra_parameters: dict[str, Any] = Field(
+        default={}, description="Additional keyword parameters passed to connect() method verbatim."
+    )
+
+    @field_validator("isolation_level")
+    @classmethod
+    def check_isolation_level(cls, v: str) -> str:
+        allowed = {None, "READ_COMMITTED", "READ_UNCOMMITTED", "REPEATABLE_READ", "SERIALIZABLE"}
```

Comment:
Seems like these are an enum.

Reply:
Hmm, round-tripping pydantic enums through JSON gives me problems, I'll keep the strings for now.